cellprofiler_core/utilities/image.py

```diff
@@ -301,16 +304,41 @@ def download_to_temp_file(url):
             Params={'Bucket': bucket_name, 'Key': key.replace("+", " ")}
         )
 
-    from urllib.request import urlopen
-    logging.info(f"Downloading image from {url}")
-    src = urlopen(url)
-    dest_file = tempfile.NamedTemporaryFile(suffix=ext, delete=False, dir=CP_TEMP_DIR.name)
-    try:
-        shutil.copyfileobj(src, dest_file)
-    except Exception as e:
-        logging.error(f"Unable to download image to temp file. {e}")
-        return None
-    finally:
-        dest_file.close()
-    return dest_file.name
+    elif scheme == 'gs':
+            # Get default Google Cloud Storage project ID from environment.
+            if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
+                with open(os.environ['GOOGLE_APPLICATION_CREDENTIALS'], 'r') as fp:
+                    # Get default project ID from Google Cloud Storage application default credential.
+                    credentials = json.load(fp)
+                    project_id = credentials['project_id']
+                    # Create client to access
+                    client = storage.Client(project_id)
```

Comment:
I think the project id parameter may not be necessary. It was not needed for the code in cellprofiler_core/readers/gcs_reader.py

This code could be updated to be more similar to the version in cellprofiler_core/readers/gcs_reader.py (e.g., no need to read the credential).